### PR TITLE
feat: add configurable legacy API limits

### PR DESCRIPTION
**Pull request by muXxer**

### CODE REVIEW

File: api/src/models/api/INetworkGetResponse.ts [hunk 1]
```diff
@@ -75,4 +75,8 @@ interface INetworkView {
      * If Identity Resolver tool should be supported.
      */
     identityResolverEnabled?: boolean;
+    /**
+     * Max results for API requests.
+     */
+    apiMaxResults?: number;
```
Comment (msarcev) [inline api/src/models/api/INetworkGetResponse.ts:81]: Should we call it `legacyApiMaxResults` or something ? 😅

File: api/src/utils/legacy/legacyTangleHelper.ts [hunk 2]
```diff
@@ -41,17 +41,22 @@ export class LegacyTangleHelper {
...
+                maxresults: maxResults,
             });
 
             if (response?.txHashes && response.txHashes.length > 0) {
                 hashes = response.txHashes;
                 cursor.node = hashes.length;
 
                 if (limit === undefined) {
-                    cursor.hasMore = hashes.length === 5000;
+                    cursor.hasMore = hashes.length === maxResults;
```
Comment (msarcev) [inline api/src/utils/legacy/legacyTangleHelper.ts:59]: You sure about this logic. The "old" line looks weird with the specific number (5000). I guess the reasoning of it is that if it's a "full page" from the node, it means there's more (?).
Are you sure the new configurable thing fits into this logic ?

File: api/src/models/api/INetworkGetResponse.ts (hunk 1 above)
Comment (begonaalvarezd) [reply api/src/models/api/INetworkGetResponse.ts:81]: we can leave it as is, but if we do we have to add to the description that for now this is implemented in legacy networks only

File: api/src/utils/legacy/legacyTangleHelper.ts (hunk 2 above)
Comment (muXxer) [reply api/src/utils/legacy/legacyTangleHelper.ts:59]: the backend doesn't have pagination. we increased the limit in the backend now to 10000, I guess when there is more than that, it's useless to display more anyway :D

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**